### PR TITLE
several UI improvements for dialogs, video messages, and the previewer activity

### PR DESCRIPTION
**Pull request by dkanada**
So the first change was fixing the padding and other spacing issues on a lot of dialogs. I probably missed a few but at least the most commonly used ones are fixed. The font preference wasn't displaying the currently selected font size, it is working now.

I also implemented the play icon I was talking about instead of using the filetype icons for image and video messages. This is what the iOS client does, and I also think it makes more sense for end users.

The last changes were in the MediaPreviewerActivity, which is now much more functional with some modifications to the glide calls. Small images and gifs are properly scaled to fill the screen, images with a transparent background are working, gifs will now play instead of just showing a thumbnail, and the UI looks more polished.

### CODE REVIEW

**Comment by bmarty** [comment]
Hi @dkanada ,

Thanks for this new PR.

Can you please sign-off and touch `CHANGES.rst` please?

Also, CI report this error: `❌ Error: 'android\:text=\"[^@]' detected 17 times, last in file './vector/src/main/res/layout/dialog_phone_number_verification.xml'.`

Why have you remove the `tools` namespace for example in a XML layout file? You can get more info on this here: https://developer.android.com/studio/write/tool-attributes

I will review the code soon.
Benoît

File: vector/src/main/java/im/vector/activity/MediaPreviewerActivity.java [hunk 1]
```diff
@@ -103,22 +108,30 @@ public void onMediaMessagePreviewClicked(@NonNull final RoomMediaMessage roomMed
...
         mPreviewerVideoView.pause();
         mFileNameView.setText(roomMediaMessage.getFileName(this));
         final String mimeType = roomMediaMessage.getMimeType(this);
         final Uri uri = roomMediaMessage.getUri();
+
+        RequestOptions options = new RequestOptions();
+        options.fitCenter();
         if (mimeType != null) {
-            if (mimeType.startsWith("image")) {
+            if (mimeType.startsWith("image") && !mimeType.contains("gif")) {
```
Comment (ganfra) [inline vector/src/main/java/im/vector/activity/MediaPreviewerActivity.java]: Thanks for your code. Normally Glide is supposed to work with .gif without specifying it is one (see SplashActivity). Might be an issue with the lib.
You might refactor your code like that:
```
final RequestOptions requestOptions = new RequestOptions().fitCenter();
        final RequestBuilder<? extends Drawable> requestBuilder;
        if (mimeType.startsWith("image")) {
            if (mimeType.contains("gif")) {
                requestBuilder = Glide.with(this).asGif();
            } else {
                requestBuilder = Glide.with(this).asDrawable();
            }
            mPreviewerImageView.setVisibility(View.VISIBLE);
            mPreviewerVideoView.setVisibility(View.GONE);
            requestBuilder
                    .apply(requestOptions)
                    .load(uri)
                    .into(mPreviewerImageView);
        } else if (mimeType.startsWith("video")) {...}
```

**Comment by dkanada** [comment]
Signed-off-by: Dillon Kanada <dkanada@protonmail.com>

File: vector/src/main/java/im/vector/adapters/VectorMessagesAdapter.java [hunk 2]
```diff
@@ -1430,43 +1430,30 @@ private View getImageVideoView(int type, final int position, View convertView, V
...
                 message = stickerMessage;
             }
 
-            // display a type watermark
-            final ImageView imageTypeView = convertView.findViewById(R.id.messagesAdapter_image_type);
-
-            if (null == imageTypeView) {
+            // display a play icon for video content
+            final ImageView playCircleView = convertView.findViewById(R.id.messagesAdapter_play_circle);
+            playCircleView.setVisibility(View.GONE);
```
Comment (bmarty) [inline vector/src/main/java/im/vector/adapters/VectorMessagesAdapter.java]: this will crash if playCircleView is null. Please to this after the next if (as it was done before)

File: vector/src/main/java/im/vector/activity/MediaPreviewerActivity.java [hunk 3]
```diff
@@ -93,8 +100,6 @@ public void onClick() {
         finish();
     }
 
-    //region MediaPreviewAdapter.EventListener
```
Comment (bmarty) [inline vector/src/main/java/im/vector/activity/MediaPreviewerActivity.java]: Why do you remove the `region` marker? @ganfra likes them, it's useful in Android Studio. Please revert

File: vector/src/main/java/im/vector/activity/MediaPreviewerActivity.java (hunk 1 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/activity/MediaPreviewerActivity.java]: Have you checked that it's still working? @ganfra has made some test and it was not working.

File: vector/src/main/java/im/vector/util/ExternalApplicationsUtil.kt [hunk 4]
```diff
@@ -84,18 +84,12 @@ fun openFileSelection(activity: Activity,
                       fragment: Fragment?,
                       allowMultipleSelection: Boolean,
                       requestCode: Int) {
-    val fileIntent = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
-        Intent(Intent.ACTION_OPEN_DOCUMENT)
-    } else {
-        Intent(Intent.ACTION_GET_CONTENT)
-    }
-
+    val fileIntent = Intent(Intent.ACTION_GET_CONTENT)
```
Comment (bmarty) [inline vector/src/main/java/im/vector/util/ExternalApplicationsUtil.kt]: was it necessary to change this? Maybe to fix a crash (#2573)?

File: vector/src/main/java/im/vector/activity/VectorHomeActivity.java [hunk 5]
```diff
@@ -1448,14 +1448,18 @@ public void onUnexpectedError(final Exception e) {
...
         LayoutInflater inflater = LayoutInflater.from(this);
-
-        AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(this);
         View dialogView = inflater.inflate(R.layout.dialog_join_room_by_id, null);
-        alertDialogBuilder.setView(dialogView);
 
         final EditText textInput = dialogView.findViewById(R.id.join_room_edit_text);
         textInput.setTextColor(ThemeUtils.INSTANCE.getColor(this, R.attr.riot_primary_text_color));
 
+        AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(this);
```
Comment (bmarty) [inline vector/src/main/java/im/vector/activity/VectorHomeActivity.java]: So I think the alertDialogBuilder variable is not necessary anymore. Can you remove it please?

File: vector/src/main/java/im/vector/activity/VectorRoomActivity.java [hunk 6]
```diff
@@ -3615,16 +3615,14 @@ private void onRoomTitleClick() {
         }
 
         LayoutInflater inflater = LayoutInflater.from(this);
+        View dialogView = inflater.inflate(R.layout.dialog_base_edit_text, null);
 
         AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(this);
+        alertDialogBuilder
```
Comment (bmarty) [inline vector/src/main/java/im/vector/activity/VectorRoomActivity.java]: Please remove this line and the `;` in the previous line

File: vector/src/main/java/im/vector/activity/VectorRoomActivity.java [hunk 7]
```diff
@@ -3682,16 +3680,14 @@ private void onRoomTopicClick() {
...
```
Comment (bmarty) [inline vector/src/main/java/im/vector/activity/VectorRoomActivity.java]: same remark

File: vector/src/main/java/im/vector/fragments/VectorMessageListFragment.java [hunk 8]
```diff
@@ -433,7 +433,7 @@ public void onE2eIconClick(final Event event, final MXDeviceInfo deviceInfo) {
         if (null != event.getCryptoError()) {
             decryptionErrorLabelTextView.setVisibility(View.VISIBLE);
             textView.setVisibility(View.VISIBLE);
-            textView.setText("**" + event.getCryptoError().getLocalizedMessage() + "**");
+            textView.setText(event.getCryptoError().getLocalizedMessage());
```
Comment (bmarty) [inline vector/src/main/java/im/vector/fragments/VectorMessageListFragment.java]: Please revert this change. We need to keep consistency with Riot iOS and Riot web

File: vector/src/main/java/im/vector/util/VectorUtils.java [hunk 9]
```diff
@@ -699,60 +693,37 @@ public static String getApplicationVersion(final Context context) {
...
-                        mMainAboutDialog.dismiss();
-                    } catch (Exception e) {
-                        Log.e(LOG_TAG, "## displayThirdPartyLicenses() : " + e.getMessage(), e);
-                    }
-                }
-                mMainAboutDialog = null;
+    private static void displayInWebView(final Context context, String url) {
+        WebView wv = new WebView(context);
+        wv.loadUrl(url);
+        wv.setWebViewClient(new WebViewClient() {
```
Comment (bmarty) [inline vector/src/main/java/im/vector/util/VectorUtils.java]: Is this WebViewClient necessary? It seems useless.

File: vector/src/main/java/im/vector/util/VectorUtils.java [hunk 10]
```diff
@@ -699,60 +693,37 @@ public static String getApplicationVersion(final Context context) {
...
      *
      * @param context the application context
      * @param url     the url to open
+     * @param title   title of the dialog
      */
-    private static void displayInWebview(final Context context, String url) {
+    private static void displayInTitleWebView(final Context context, String url, String title) {
         WebView wv = new WebView(context);
         wv.loadUrl(url);
         wv.setWebViewClient(new WebViewClient() {
```
Comment (bmarty) [inline vector/src/main/java/im/vector/util/VectorUtils.java]: Maybe this one is not necessary as well.

File: vector/src/main/res/layout/dialog_phone_number_verification.xml [hunk 11]
```diff
@@ -1,10 +1,8 @@
 <?xml version="1.0" encoding="utf-8"?>
 <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    android:orientation="vertical"
-    android:padding="16dp">
+    android:orientation="vertical">
```
Comment (bmarty) [inline vector/src/main/res/layout/dialog_phone_number_verification.xml]: No padding for this one?

File: vector/src/main/res/layout/dialog_ssl_fingerprint.xml [hunk 12]
```diff
@@ -1,53 +1,45 @@
 <?xml version="1.0" encoding="utf-8"?>
-
 <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
-              android:orientation="vertical"
-              android:layout_width="match_parent"
-              android:layout_height="match_parent">
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical">
```
Comment (bmarty) [inline vector/src/main/res/layout/dialog_ssl_fingerprint.xml]: no padding for this one?

File: vector/src/main/res/values/strings.xml [hunk 13]
```diff
@@ -874,7 +874,7 @@
 
     <string name="encryption_information_verify_device">Verify device</string>
     <string name="encryption_information_verify_device_warning">To verify that this device can be trusted, please contact its owner using some other means (e.g. in person or a phone call) and ask them whether the key they see in their User Settings for this device matches the key below:</string>
-    <string name="encryption_information_verify_device_warning2">If it matches, press the verify button below. \nIf it doesn\'t, then someone else is intercepting this device and you should probably blacklist it.\nIn future this verification process will be more sophisticated.</string>
+    <string name="encryption_information_verify_device_warning2">If it matches, press the verify button below. If it doesn\'t, then someone else is intercepting this device and you should probably blacklist it. In the future this verification process will be more sophisticated.</string>
```
Comment (bmarty) [inline vector/src/main/res/values/strings.xml]: Why have you delete all the `\n` ?

File: vector/src/main/java/im/vector/activity/MediaPreviewerActivity.java (hunk 1 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/activity/MediaPreviewerActivity.java]: Just checked now and it's working well

File: vector/src/main/res/values/strings.xml (hunk 13 above)
Comment (dkanada) [reply vector/src/main/res/values/strings.xml]: It looked kind of odd with only a single newline, I figured either two newlines or zero newlines would work better. I can revert it if you don't like the change.

File: vector/src/main/res/layout/dialog_phone_number_verification.xml (hunk 11 above)
Comment (dkanada) [reply vector/src/main/res/layout/dialog_phone_number_verification.xml]: Didn't have a way to view this dialog, I can add padding but I won't be able to test.

File: vector/src/main/res/layout/dialog_ssl_fingerprint.xml (hunk 12 above)
Comment (dkanada) [reply vector/src/main/res/layout/dialog_ssl_fingerprint.xml]: Didn't have a way to view this dialog, I can add padding but I won't be able to test.

File: vector/src/main/java/im/vector/activity/VectorHomeActivity.java (hunk 5 above)
Comment (dkanada) [reply vector/src/main/java/im/vector/activity/VectorHomeActivity.java]: It is actually used later to add some behavior to a button, same for all the other cases.

File: vector/src/main/java/im/vector/adapters/VectorMessagesAdapter.java (hunk 2 above)
Comment (dkanada) [reply vector/src/main/java/im/vector/adapters/VectorMessagesAdapter.java]: Done.

File: vector/src/main/java/im/vector/fragments/VectorMessageListFragment.java (hunk 8 above)
Comment (dkanada) [reply vector/src/main/java/im/vector/fragments/VectorMessageListFragment.java]: Done.

File: vector/src/main/java/im/vector/util/VectorUtils.java (hunk 10 above)
Comment (dkanada) [reply vector/src/main/java/im/vector/util/VectorUtils.java]: Removed both web view clients and didn't see any issues.

Comment (bmarty) [reply vector/src/main/java/im/vector/util/VectorUtils.java]: ok, thanks

File: vector/src/main/java/im/vector/util/ExternalApplicationsUtil.kt (hunk 4 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/util/ExternalApplicationsUtil.kt]: ok

File: vector/src/main/res/values/strings.xml (hunk 13 above)
Comment (bmarty) [reply vector/src/main/res/values/strings.xml]: ok, let's keep your change for the moment
